src/http/headers.cr

```diff
@@ -9,13 +9,12 @@ struct HTTP::Headers
   record Key, name : String do
     forward_missing_to @name
 
-    def hash
-      h = 0
-      name.each_byte do |c|
-        c = normalize_byte(c)
-        h = 31 * h + c
+    def hash(hasher)
+      hasher.raw(bytesize.to_u32)
```

Comment:
Why raw here? I think that the correct way to use hasher should *always* be `<<` without anyone having to think about `raw` except people implementing `hash(hasher)` on numbers. Shouldn't `raw` be `:nodoc:`?